upgrades/tests/test_node.py

```diff
@@ -0,0 +1,143 @@
+import pytest
+import unittest.mock as mock
+
+import upgrade_testing_framework.cluster_management.node as node
+
+def test_WHEN_create_NodeConfiguration_THEN_has_expected_values():
+    # Set up our test
+    test_node_name = "node-name"
+    test_cluster_name = "cluster-name"
+    test_master_nodes = ["m1", "m2"]
+    test_seed_hosts = ["s1", "s2"]
+
+    # Run our test
+    actual_value = node.NodeConfiguration(test_node_name, test_cluster_name, test_master_nodes, test_seed_hosts)
+
+    # Check the results
+    expected_value = {
+        "cluster.name": test_cluster_name,
+        "cluster.initial_master_nodes": ",".join(test_master_nodes),
+        "discovery.seed_hosts": ",".join(test_seed_hosts),
+        "node.name": test_node_name,
+        "bootstrap.memory_lock": "true",
+        "ES_JAVA_OPTS": "-Xms512m -Xmx512m"
+    }
+    assert expected_value == actual_value.config
+
+def test_WHEN_node_start_THEN_as_expected():
```

Comment:
A lot of these cases are happy cases, which I am assuming is temporary and will change once we start tackling some edge cases and add to this

Reply:
Yeah, that's the plan.  I want to do that in either the next revision (in-progress) or the one after.  TBH though, trying to work backwards from need so it may just happen the first time one of us runs into a problem with it.